libgnucash/engine/SX-ttinfo.cpp

```diff
@@ -22,337 +22,235 @@
+static void delete_splitinfo (TTSplitInfo* tt_info);
+
+struct TTInfo_s
+{
+    OptionalString description;
+    OptionalString num;
+    OptionalString notes;
+    gnc_commodity *common_currency = nullptr;
+    TTSplitInfoVec splits;
+    ~TTInfo_s () { std::for_each (splits.begin(), splits.end(), delete_splitinfo); };
 };
 
 struct TTSplitInfo_s
 {
-    char *action; /* owned by us */
-    char *memo; /* owned by us */
-    char *credit_formula, *debit_formula; /* owned by us */
-    Account *acc;
+    OptionalString action;
+    OptionalString memo;
+    OptionalString credit_formula, debit_formula;
+    Account *acc = nullptr;
 };
 
 TTInfo *
 gnc_ttinfo_malloc(void)
 {
-    TTInfo *tti = g_new0(TTInfo, 1);
-    return tti;
+    return new TTInfo;
 }
 
 static void
-delete_splitinfo(gpointer data, gpointer user_data)
+delete_splitinfo (TTSplitInfo* tt_info)
 {
-    gnc_ttsplitinfo_free( (TTSplitInfo *) data);
-    return;
+    delete tt_info;
 }
 
 void gnc_ttinfo_free(TTInfo *info)
 {
     g_return_if_fail(info);
-
-    g_free(info->description);
-    g_free(info->num);
-    g_free (info->notes);
-    g_list_foreach(info->splits,
-                   delete_splitinfo,
-                   NULL);
-
-    g_list_free(info->splits);
-
-    g_free(info);
-
-    return;
+    delete info;
 }
 
 void
 gnc_ttinfo_set_description(TTInfo *tti, const char *description)
 {
     g_return_if_fail(tti);
-
-    if (tti->description)
-    {
-        g_free(tti->description);
-    }
-
-    tti->description = g_strdup(description);
-
-    return;
+    tti->description = description;
 }
 
 const char *
 gnc_ttinfo_get_description(TTInfo *tti)
 {
     g_return_val_if_fail(tti, NULL);
-
-    return tti->description;
+    return *tti->description;
 }
 
-
-
-
 void
 gnc_ttinfo_set_num(TTInfo *tti, const char *num)
 {
     g_return_if_fail(tti);
-
-    if (tti->num)
-    {
-        g_free(tti->num);
-    }
-
-    tti->num = g_strdup(num);
-
-    return;
+    tti->num = num;
 }
 
 const char*
 gnc_ttinfo_get_num(TTInfo *tti)
 {
     g_return_val_if_fail(tti, NULL);
-
-    return tti->num;
+    return *tti->num;
 }
 
 void
 gnc_ttinfo_set_notes (TTInfo *tti, const char *notes)
 {
     g_return_if_fail (tti);
-
-    if (tti->notes)
-    {
-        g_free (tti->notes);
-    }
-
-    tti->notes = g_strdup (notes);
-
-    return;
+    tti->notes = notes;
 }
 
 const char*
 gnc_ttinfo_get_notes (TTInfo *tti)
 {
     g_return_val_if_fail (tti, NULL);
-
-    return tti->notes;
+    return *tti->notes;
 }
 
 void
 gnc_ttinfo_set_currency(TTInfo *tti, gnc_commodity *common_currency)
 {
     g_return_if_fail(tti);
-
     tti->common_currency = common_currency;
-    return;
 }
 
 gnc_commodity *
 gnc_ttinfo_get_currency(TTInfo *tti)
 {
     g_return_val_if_fail(tti, NULL);
-
     return tti->common_currency;
 }
 
 
-void gnc_ttinfo_set_template_splits(TTInfo *tti, GList *splits)
+void gnc_ttinfo_set_template_splits(TTInfo *tti, TTSplitInfoVec splits)
```

Comment:
`TTSplitInfoVec&& splits` to move the `TTSplitInfoVec` created in `sxftd_add_template_trans` or better still replace this with `TTInfo::add_template_split(Split*)`.